src/ConnectedMode/Suppressions/ServerSuppressionsChangedHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core.Suppression;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Suppressions
+{
+    [Export(typeof(ServerSuppressionsChangedHandler))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ServerSuppressionsChangedHandler : IDisposable
+    {
+        private readonly IClientSuppressionSynchronizer clientSuppressionSynchronizer;
+        private readonly ISuppressedIssuesMonitor suppressedIssuesMonitor;
+
+        [ImportingConstructor]
+        public ServerSuppressionsChangedHandler(IClientSuppressionSynchronizer clientSuppressionSynchronizer, ISuppressedIssuesMonitor suppressedIssuesMonitor)
+        {
+            this.clientSuppressionSynchronizer = clientSuppressionSynchronizer;
+            this.suppressedIssuesMonitor = suppressedIssuesMonitor;
+
+            suppressedIssuesMonitor.SuppressionsUpdateRequested += ServerSuppressionsChanged;
+        }
+
+        private void ServerSuppressionsChanged(object sender, EventArgs e)
+        {
+            clientSuppressionSynchronizer.SynchronizeSuppressedIssues();
```

Comment:
Note here to maybe add logging

Reply:
Either here, or in the synchronizer (or perhaps both).
We need to decide what information would be useful in normal operation (e.g. "_Re-apply server-side suppressions_") and what would help in troubleshooting/support cases (e.g. "x issues suppressed/updated").
I'll add a task to the board.